### PR TITLE
Semantic Snippets - Remove reflection and call the LanguageServerSnippetExpander directly

### PR DESCRIPTION
**Pull request by akhera99**
Needed to update the FoldingRangeTests to account for changes from FoldingRangeKind being an enum to being a struct. 

Need to remove the TestRoslynLanguageServerSnippetExpander as well as the RoslynLSPSnippetExpander files as they are no longer necessary.

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 1]
```diff
@@ -39,6 +42,7 @@
     <PackageReference Include="Microsoft.VisualStudio.Language" Version="$(MicrosoftVisualStudioLanguageVersion)" />
     <!-- No warn on NU1701 until the LSP client targets netstandard2.0 https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1369985/ -->
     <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client" Version="$(MicrosoftVisualStudioLanguageServerClientVersion)" PrivateAssets="all" NoWarn="NU1701" />
+    <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client.Implementation" Version="$(MicrosoftVisualStudioLanguageServerClientImplementationVersion)" PrivateAssets="all" NoWarn="NU1701" />
```
Comment (davidwengier) [inline src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:45]: @gundermanc Does this pose a problem for VS Mac? Currently Microsoft.VisualStudio.LanguageServer.Client.Implementation is excluded from the VS Mac MEF composition, but I'm not sure if thats due to technical limitations or just history.

Comment (davidwengier) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:45]: Asked in https://github.com/dotnet/roslyn/pull/61496 instead

Comment (gundermanc) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:45]: > Does this pose a problem for VS Mac? 

It works but some care must be taken:

- The same DLL exists on VS Mac but some of the surface area and/or implementations are slightly different.
- This DLL is ugly dependency-wise. You just need it for the public API but unfortunately receive transitive dependencies on VS Windows types and potentially WPF.

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 2]
```diff
@@ -9,6 +9,9 @@
     <DefineConstants>$(DefineConstants);EDITOR_FEATURES</DefineConstants>
     <ApplyNgenOptimization Condition="'$(TargetFramework)' == 'netstandard2.0'">full</ApplyNgenOptimization>
 
+    <!-- CA1416 can go away when we target net6.0-macos -->
+    <NoWarn>$(NoWarn);NU1701;</NoWarn>
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:13]: I had to deal with similar in #61234 and can help update this.

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (gundermanc) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:45]: Unfortunately, this code will probably also eventually be needed in VS Mac.

[the remark this answers is not in the document]
Comment (davidwengier) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:45]: I think moving to EditorFeatures.WPF makes sense, even if we have to mirror the change in EditorFeatures.Cocoa (and ideally that would be done at the same time), as it prevents someone else from later adding some new usage of a Windows-only aspect of the implementation assembly without realizing it.

**Comment by JoeRobich** [comment]
The failing MacOS test has previously been reported as flaky - https://github.com/dotnet/roslyn/issues/61474

File: src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs [hunk 3]
```diff
@@ -39,7 +38,7 @@ internal sealed class CommitManager : IAsyncCompletionCommitManager
         private readonly ITextView _textView;
         private readonly IGlobalOptionService _globalOptions;
         private readonly IThreadingContext _threadingContext;
-        private readonly RoslynLSPSnippetExpander _roslynLSPSnippetExpander;
+        private readonly ILanguageServerSnippetExpander? _languageServerSnippetExpander;
```
Comment (jmarolf) [inline src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs:41]: Why is this nullable? won't this always be imported successfully if the VS version is new enough?

Comment (sharwell) [reply src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs:41]: No, it is exported by a higher layer (EditorFeatures is WPF+Cocoa, but this is only exported for WPF).

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 4]
```diff
@@ -37,13 +37,7 @@
          There is a test to ensure this doesn't regress in src\EditorFeatures\Test\AssemblyReferenceTests.cs -->
     <PackageReference Include="Microsoft.VisualStudio.ImageCatalog" Version="$(MicrosoftVisualStudioImageCatalogVersion)" PrivateAssets="all" />
     <PackageReference Include="Microsoft.VisualStudio.Language" Version="$(MicrosoftVisualStudioLanguageVersion)" />
-    <!-- No warn on NU1701 until the LSP client targets netstandard2.0 https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1369985/ -->
-    <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client" Version="$(MicrosoftVisualStudioLanguageServerClientVersion)" PrivateAssets="all" NoWarn="NU1701" />
-    <!-- Explicit reference to Shell.15.0 et. al. with NU1701 only until the LSP client targets netstandard2.0 https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1369985/ -->
-    <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="$(MicrosoftVisualStudioShell150Version)" PrivateAssets="all" ExcludeAssets="all" NoWarn="NU1701" />
```
Comment (jmarolf) [inline src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:43]: u love to see it!

**Review by jmarolf — APPROVED** [review]
:shipit:

**Pull request merged**